File: F/2833-furthest-point-from-origin/main.go
func furthestDistanceFromOrigin(moves string) int {
	cnt := 0
	sum := 0
	for _, b := range []byte(moves) {
		if b == 'R' {
			sum++
		} else if b == 'L' {
			sum--
		} else {
			cnt++
		}
	}

	if sum < 0 {
		sum = -sum
	}
	return cnt + sum
}

File: F/2833-furthest-point-from-origin/main.rs
impl Solution {
  pub fn furthest_distance_from_origin(moves: String) -> i32 {
    let (sum, spaces) = moves.as_bytes().iter().fold((0 as i32, 0), |(sum, spaces), &b| {
      (
        sum
          + if b == b'L' {
            1
          } else if b == b'R' {
            -1
          } else {
            0
          },
        spaces + if b == b'_' { 1 } else { 0 },
      )
    });
    sum.abs() + spaces
  }
}
